locator/abstract_replication_strategy.hh

```diff
@@ -128,11 +128,48 @@ public:
     future<dht::token_range_vector> get_pending_address_ranges(const token_metadata_ptr tmptr, std::unordered_set<token> pending_tokens, inet_address pending_address, locator::endpoint_dc_rack dr) const;
 };
 
+using replication_strategy_ptr = seastar::shared_ptr<const abstract_replication_strategy>;
+using mutable_replication_strategy_ptr = seastar::shared_ptr<abstract_replication_strategy>;
+
+/// \brief Represents effective replication (assignment of replicas to keys).
+///
+/// It's a result of application of a given replication strategy
+/// over a particular token metadata version, for a given table.
```

Comment:
using a given set of replication strategy options.

Reply:
not to confuse with a "pure" replication strategy which is a synonym for the class/type.